### PR TITLE
Refactor `ScalarType`

### PR DESCRIPTION
**Pull request by sav007**
Replace `javaType(): Class<*>` with `className(): String` for better support of KN

### CODE REVIEW

**Comment by tasomaniac** [comment]
I had a quick look but didn't continue since this needs rebasing to master. I guess keeping the new implementation in commonMain sourceSet would be fine.

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt [hunk 1]
```diff
@@ -1,10 +1,117 @@
...
+    } as CustomTypeAdapter<T>
+  }
 
   companion object {
-    val DEFAULT: ScalarTypeAdapters
+    @JvmField
+    val DEFAULT = ScalarTypeAdapters(emptyMap())
+
+    private val DEFAULT_ADAPTERS = emptyMap<String, CustomTypeAdapter<*>>() +
+        createDefaultScalarTypeAdapter("java.lang.String", "kotlin.String") { value ->
```
Comment (tasomaniac) [inline apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt]: Is `java.lang.String` enough or should we also check other Strings? Like I wonder what would happen on iOS. Of course can be tackled later.

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt [hunk 2]
```diff
@@ -1,10 +1,117 @@
...
+        createDefaultScalarTypeAdapter("java.lang.Double", "kotlin.Double") { value ->
+          when (value) {
+            is GraphQLNumber -> value.value.toDouble()
+            is GraphQLString -> value.value.toDouble()
+            else -> throw IllegalArgumentException("Can't decode: $value into Double")
+          }
+        } +
+        createDefaultScalarTypeAdapter(FileUpload::class.qualifiedName!!) { value ->
+          when (value) {
+            is GraphQLNumber -> value.value.toDouble()
```
Comment (tasomaniac) [inline apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt]: Why does this say Double for FileUpload?

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt (hunk 1 above)
Comment (sav007) [reply apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt]: on iOS it must be `kotlin.String` as generated code is platform agnostic (common).

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt (hunk 2 above)
Comment (sav007) [reply apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt]: ogh, jeez my bad

File: apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt (hunk 1 above)
Comment (tasomaniac) [reply apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ScalarTypeAdapters.kt]: 👍 sounds good.

**Pull request merged**